x/gov/client/cli/tx.go

```diff
@@ -131,9 +131,11 @@ $ gaiacli tx gov deposit 1 10stake --from mykey
 			}
 
 			// check to see if the proposal is in the store
-			_, err = govClientUtils.QueryProposalByID(proposalID, cliCtx, cdc, queryRoute)
-			if err != nil {
-				return fmt.Errorf("Failed to fetch proposal-id %d: %s", proposalID, err)
+			if !cliCtx.GenerateOnly {
```

Comment:
tbh, I don't even think we should do this. The message handler should return an error if the proposal ID doesn't exist. Let's gut this entire code block.

Reply:
I agree. Removed